x-pack/filebeat/docs/inputs/input-unifiedlogs.asciidoc

```diff
@@ -0,0 +1,183 @@
+[role="xpack"]
+
+:type: unifiedlogs
+
+[id="{beatname_lc}-input-{type}"]
+=== Unified Logs input
+
+++++
+<titleabbrev>Unified Logs</titleabbrev>
+++++
+
+NOTE: Only available for MacOS.
+
+The unified logging system provides a comprehensive and performant API to capture
+telemetry across all levels of the system. This system centralizes the storage of
+log data in memory and on disk, rather than writing that data to a text-based log file.
+
+The input interacts with the `log` command-line tool to provide access to the events.
+It can process events in some different ways:
+
+- Streaming new, incoming events. This is the default.
+- Start processing events from a start date. This happens when the input is restarted,
+or when a specific start date is provided.
+- Start processing events since the beginning of time, by setting the `backill` option.
```

Comment:
Do I understand it correclty that the input:

* does not use the persistent state storage (registry)
* would skip events when restarted in its default configuration

I'm concerned that basically the default configuration allows data loss and our products give the "at least once delivery guarantee". This input seems to break this promise and I think our users would expect it not to.

Would it be better to store the timestamp offset in the registry (like we do with file offsets in filestream) and continue from where we left off after the input restart?

Reply:
The input uses the registry to keep track of the last event timestamp.

It will start from where it stopped. It does not have data loss, but on the contrary on restarts some data might be reprocessed since the resolution of the underlying tool is up to the second, so some events might overlap and be re-sent.

If this is not clear from the documentation, we can work on some rephrasing if there are any confusing bits.

Reply:
I guess this was not clear for me from the

> Streaming new, incoming events. This is the default.

As a user, I'd think that on every restart it start streaming from the current moment, not from where it left off.

I'd add some additional explanation what streaming incoming event means or perhaps a separate section that explains that we always track the last timestamps and continue on restart.

Reply:
That was referring to the initial moment, a user can choose to backfill old data or star only with new events. In the next line this is clarified:

> - Streaming new, incoming events. This is the default.
> - Start processing events from a start date. This happens when the input is restarted,
or when a specific start date is provided.

Reply:
But before these 3 bullet points we have:

> It can process events in some different ways:

Normally, that would mean 3 different modes this input can operate in. I understood it as:

> I can configure the input to operate in "stream from now" mode, "stream from a given date" mode and "stream from beginning of time" mode.

So, what's the default behavior of the input then? I can't find a description where it would say something like:

> By default, the input starts streaming events from the current point of time but when restarted will continue where it left off.

It's up to you whether we need to change the docs or not, this is just my personal feedback.

Reply:
Makes sense now, updated the docs 👍